devito/finite_differences/differentiable.py

```diff
@@ -1005,3 +1006,28 @@ def _diff2sympy(obj):
 evalf_table[Add] = evalf_table[sympy.Add]
 evalf_table[Mul] = evalf_table[sympy.Mul]
 evalf_table[Pow] = evalf_table[sympy.Pow]
+
+
+# Interpolation for finite differences
+@singledispatch
+def interp_for_fd(expr, x0, **kwargs):
+    return expr
+
+
+@interp_for_fd.register(AbstractFunction)
+def _(expr, x0, **kwargs):
+    from devito.finite_differences.derivative import Derivative
+    x0_expr = {d: v for d, v in x0.items() if v is not expr.indices_ref[d]}
+    if x0_expr and not expr.is_parameter:
+        dims = tuple((d, 0) for d in x0_expr)
+        fd_o = tuple([2]*len(dims))
+        return Derivative(expr, *dims, fd_order=fd_o, x0=x0_expr)._evaluate(**kwargs)
+    else:
+        return expr
+
+
+@interp_for_fd.register(Add)
```

Comment:
why not just the default behaviour? in the absence of args return expr otherwise this?

sounds safer

Reply:
Made general to `sympy.Expr`